analyzers/tests/SonarAnalyzer.Test/TestCases/NamedPlaceholdersShouldBeUnique.cs

```diff
@@ -0,0 +1,179 @@
+
+            logger.LogInformation("Hey {foo} and {foo,42} and {foo:format} and {foo,-42:for_{_mat}", foo, bar);
+            //                                    ^^^ {{Message template placeholder 'foo' is not unique.}}
+            //                                                 ^^^ @-1 {{Message template placeholder 'foo' is not unique.}}
+            //                                                                  ^^^ @-2 {{Message template placeholder 'foo' is not unique.}}
+
+            logger.LogInformation("Hey {_foo} and {_foo42} and {_foo42} and {_foo}", foo, bar);
+            //                                                  ^^^^^^ {{Message template placeholder '_foo42' is not unique.}}
+            //                                                               ^^^^ @-1 {{Message template placeholder '_foo' is not unique.}}
+
+            // Multiline
+            logger.LogDebug(
+                message: "Hey {foo} and {foo}", // Noncompliant
+                //                       ^^^
+                eventId: eventId,
+                exception: ex,
+                args: new[] { foo, foo });
+
+            LoggerExtensions.Log(
+                logger,
+                logLevel: LogLevel.Trace,
+                args: Array.Empty<object>(),
+                message: "Hey {foo} and {foo}"); // Noncompliant
+            //                           ^^^
+
+            logger.LogWarning(@"
+                hey there
+                {foo}
+                and {bar}
+                and again {foo}
+                ",
+            //             ^^^ @-1 {{Message template placeholder 'foo' is not unique.}}
+                foo, bar, foo);
+        }
+
+        // Fake
+        public class NotLogger
+        {
+            public void LogDebug(string message, params object[] args) { }
+        }
+    }
+
+    public static class NotLoggerExtensions
+    {
+        public static void LogCritical(this Program.NotLogger logger, string message, params object[] args) { }
+    }
+}
+
+namespace SerilogTests
+{
+    using Serilog;
+    using Serilog.Events;
+
+    public class Program
+    {
+        void Compliant(ILogger logger, string arg)
+        {
+            logger.Write(LogEventLevel.Information, "Hey {foo} and {bar}");                // Compliant
+            logger.Write(LogEventLevel.Information, "Hey {foo} and {bar}", arg);            // Compliant
+            logger.Write(LogEventLevel.Information, "Hey {foo} and {bar}", arg, arg);      // Compliant
+        }
+
+        void Noncompliant(ILogger logger, string arg)
+        {
+            logger.Write(LogEventLevel.Information, "Hey {foo} and {foo}");                // Noncompliant
+            logger.Write(LogEventLevel.Information, "Hey {foo} and {foo}", arg);            // Noncompliant
+            logger.Write(LogEventLevel.Information, "Hey {foo} and {foo}", arg, arg);      // Noncompliant
```

Comment:
Aren't these tests now included in the `NamedPlaceholdersShouldBeUnique_Serilog_CS` test?

Reply:
No, they have slightly diffent signature (require a `LogEventLevel` argument), so they mismatch the `DataRow` style of the other test. I left them here for simplicity, to not make a custom test just for this.